lib/matplotlib/widgets.py

```diff
@@ -2468,57 +2522,84 @@ def _onmove(self, event):
 
             x0, x1, y0, y1 = (center[0] - dx, center[0] + dx,
                               center[1] - dy, center[1] + dy)
+            width = x1 - x0
+            height = y1 - y0
 
-        self.extents = x0, x1, y0, y1
+        self._set_corner_width_rotation((x0, y0), width, height, rotation)
+        # Save the current mouse coordinates
+        self._prev_xmove = xmove
+        self._prev_ymove = ymove
 
     @property
-    def _rect_bbox(self):
+    def _rect_properties(self):
+        """
+        Returns
+        -------
+        x0, y0, width, height, angle (in radians)
+        """
         if self.drawtype == 'box':
             x0 = self.to_draw.get_x()
             y0 = self.to_draw.get_y()
             width = self.to_draw.get_width()
             height = self.to_draw.get_height()
-            return x0, y0, width, height
+            angle = self.to_draw.get_angle()
+            return x0, y0, width, height, np.deg2rad(angle)
         else:
             x, y = self.to_draw.get_data()
             x0, x1 = min(x), max(x)
             y0, y1 = min(y), max(y)
-            return x0, y0, x1 - x0, y1 - y0
+            return x0, y0, x1 - x0, y1 - y0, 0
 
     @property
     def corners(self):
         """Corners of rectangle from lower left, moving clockwise."""
-        x0, y0, width, height = self._rect_bbox
-        xc = x0, x0 + width, x0 + width, x0
-        yc = y0, y0, y0 + height, y0 + height
-        return xc, yc
+        x0, y0, width, height, angle = self._rect_properties
+        x = np.array([0, width, width, 0])
+        y = np.array([0, 0, height, height])
+        xc = x0 + (np.cos(angle) * x - np.sin(angle) * y)
```

Comment:
You have a ton of these rotations.  Is there a reason you don't use `mtransforms.rotate` for these?

Reply:
Nope, in fact it turns out this is just the inverse of the transform I added a method to get, so these are all much simplified now, thanks for catching.